pkg/sql/catalog/tabledesc/mutation.go

```diff
@@ -369,9 +311,10 @@ func (m mutation) MutationOrdinal() int {
 
 // mutationCache contains precomputed slices of catalog.Mutation interfaces.
 type mutationCache struct {
-	all     []catalog.Mutation
-	columns []catalog.Mutation
-	indexes []catalog.Mutation
+	all                               []catalog.Mutation
+	columns                           []catalog.Mutation
+	indexes                           []catalog.Mutation
+	checks, fks, uniqueWithoutIndexes []catalog.Mutation
```

Comment:
This change is good -- it offloads a lot of codes from constraint.go to this file when building the constraint cache!